lib/media/drm_engine.js

```diff
@@ -1870,8 +1870,27 @@ shaka.media.DrmEngine = class {
         offlineConfig.sessionTypes = ['persistent-license'];
 
         const configs = [offlineConfig, basicConfig];
-
-        const access = await navigator.requestMediaKeySystemAccess(
+        // On some (Android) WebView environments,
+        // requestMediaKeySystemAccess will
+        // not resolve or reject, at least if RESOURCE_PROTECTED_MEDIA_ID
+        // is not set.  This is a workaround for that issue.
+        const TIMEOUT_FOR_CHECKACCESS_IN_SECONDS = 1;
+        const checkAccessOrTimeout = (keySystem, configs) => {
+          return new Promise((resolve, reject) => {
+            let resolved = false;
+            navigator.requestMediaKeySystemAccess(keySystem, configs)
+                .then((access) => {
+                  resolved = true;
+                  resolve(access);
+                });
+            new shaka.util.Timer(() => {
+              if (!resolved) {
+                reject(new Error('Timeout requestMediaKeySystemAccess'));
+              }
+            }).tickAfter(TIMEOUT_FOR_CHECKACCESS_IN_SECONDS);
```

Comment:
You could add a utility method to simplify it, sth like:

```js
promiseWithTimeout(promise, timeout) {
  return Promise.race([
    promise,
    new Promise((_, reject) => new shaka.util.Timer(reject).tickAfter(timeout));
  ]);
}
```

Reply:
Added utility function promiseWithTimeout